### PR TITLE
Reset button styling and position updated to new design

### LINKED ISSUE #315: Buttons

### LINKED ISSUE #323: Reset button

### PR DESCRIPTION
**Pull request by rjmcneill**
Closes #323 
Closes #315 as well since this is the last button that needed styling

### CODE REVIEW

File: hello.patch [hunk 1]
```diff
@@ -0,0 +1,51 @@
+diff --git a/src/assets/js/chart/nav.js b/src/assets/js/chart/nav.js
```
Comment (jleft) [inline hello.patch]: Remove this file

File: src/assets/js/chart/nav.js [hunk 2]
```diff
@@ -30,9 +32,28 @@
         var layoutWidth;
 
         function nav(selection) {
-            var model = selection.datum();
+            var navbarContainer = selection.select('#navbar-container');
+            var navbarReset = selection.select('#navbar-reset');
+            var model = navbarContainer.datum();
+
+            navbarContainer.style('margin-right', '0');
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: This is overwriting a style in the Less - can we just change the Less?

File: src/assets/js/chart/nav.js [hunk 3]
```diff
@@ -30,9 +32,28 @@
...
 
             viewScale.domain(model.viewDomain);
+            navbarReset
+                .style('margin-right', '21')
+                .style('margin-left', '15')
+                .style('margin-top', '10')
+                .select('g')
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Do we need the `g` element?

File: src/assets/js/chart/nav.js [hunk 4]
```diff
@@ -30,9 +32,28 @@
...
+                .on('click', function() { dispatch[sc.event.resetToLatest](); })
+                .attr('xmlns', 'http://www.w3.org/2000/svg')
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Don't think we need this?

File: src/assets/js/chart/nav.js [hunk 5]
```diff
@@ -30,9 +32,28 @@
...
+                .classed('reset-button-inactive', model.trackingLatest)
+                .classed('reset-button-active', !model.trackingLatest)
+                .html(function() {
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Stylistically, prefer:

```js
.append('path')
.attr('d', 'M1.5 1.5h13.438L23 20.218 14.937 38H1.5l9.406-17.782L1.5 1.5z');
```

However, will this not be updating the `path` element on every render? Can we make use of an enter selection to make sure this doesn't happen? (Could be an argument for using `decorate` on cartesian chart)

File: src/assets/js/chart/nav.js [hunk 6]
```diff
@@ -30,9 +32,28 @@
...
+                    return '<path d="M1.5 1.5h13.438L23 20.218 14.937 38H1.5l9.406-17.782L1.5 1.5z"></path>';
+                });
+
+            viewScale.domain(model.viewDomain)
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: The domain is being set twice (here and on line 41) - only need to set it once?

File: src/assets/js/chart/nav.js [hunk 7]
```diff
@@ -30,9 +32,28 @@
...
+                .range([0, navbarContainer.node().getAttribute('layout-width')]);
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Don't need this - see `nav.dimensionChanged`

File: src/assets/js/chart/nav.js [hunk 8]
```diff
@@ -67,7 +88,8 @@
                     dispatch[sc.event.viewChange](domain);
                 });
 
-            selection.select('.plot-area').call(zoom);
+            navbarContainer.select('.plot-area').call(zoom);
+            navbarContainer.call(zoom);
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Only need to call zoom on the plot area?

File: src/assets/styles/navigator.less [hunk 9]
```diff
@@ -1,33 +1,52 @@
...
+      fill: @font-colour;
+    }
   }
 
-  text {
-    font-size: @font-size-base;
-    fill: @font-colour;
+  #navbar-reset {
+    stroke: @primary-white;
+    width: 24px;
```
Comment (jleft) [inline src/assets/styles/navigator.less]: If we're flex for layout, would this be more appropriate?

```less
flex: 0 0 24px;
```

File: src/assets/styles/navigator.less [hunk 10]
```diff
@@ -1,33 +1,52 @@
...
+    height: 40px;
+    opacity: .199;
```
Comment (jleft) [inline src/assets/styles/navigator.less]: Could just use `0.2`?

File: src/assets/styles/navigator.less [hunk 11]
```diff
@@ -1,33 +1,52 @@
...
+    fill-rule: evenodd;
```
Comment (jleft) [inline src/assets/styles/navigator.less]: Do we need this?

File: src/index.html.tpl [hunk 12]
```diff
@@ -37,6 +34,7 @@
                 </div>
                 <div id="navbar-row">
                     <svg id="navbar-container"></svg>
+                    <svg id="navbar-reset"><g></g></svg>
```
Comment (jleft) [inline src/index.html.tpl]: Don't think we need the `g` (again, possibly an argument for using `decorate` on cartesian chart)

File: src/assets/js/chart/nav.js (hunk 8 above)
Comment (Afredon) [inline src/assets/js/chart/nav.js]: There is also a mix-up that probably comes from rebasing on my last changes with the .call(zoom) on same line than the select and now a duplication of it and an unneeded repetition of navbarContainer either.

File: src/assets/js/chart/nav.js (hunk 7 above)
Comment (rjmcneill) [inline src/assets/js/chart/nav.js]: Sorry, these came from merging in Arnaud's stuff incorrectly

File: src/assets/styles/navigator.less (hunk 10 above)
Comment (rjmcneill) [inline src/assets/styles/navigator.less]: This is what was given to us in the icons @jleft 
Should we just make it 0.2 though?
Same with the oddeven fill rule

Comment (jleft) [inline src/assets/styles/navigator.less]: The design also specifies:

> Colour when user scrolls away from most up-to-date data = #FFFFFF fill - with 20% opacity

I'd use `0.2` in this case - I think that was the intention, based on the above comment in the designs (it's a nicer number too!).

> Same with the oddeven fill rule

Happy to use this, if it's needed. If it's not, then I think it'd be best to remove it.

**Pull request merged**